src/main/java/io/github/hypixel_api_wrapper/wrapper/player/HypixelPlayer.java

```diff
@@ -9,10 +10,14 @@ public class HypixelPlayer {
 
     private String username;
     private RequestFactory requestFactory;
+    HypixelPlayerStats stats;
 
     public HypixelPlayer(String username, RequestFactory requestFactory) {
+
         this.username = username;
         this.requestFactory = requestFactory;
+
+        stats = new HypixelPlayerStats(requestFactory.getEndpointThroughAPI(Endpoint.PLAYER));
```

Comment:
This is not dynamic loading. This should move to a separate method that then gets called inside of any other method that actually returns stats.

Reply:
We could do this, but I'm not so concerned since everything inside it is dynamically loaded.

Reply:
But now the constructor's fetching the `/player` endpoint in the constructor, even if the user doesn't care about the stats of the player itself. This:
```java
requestFactory.getEndpointThroughAPI(Endpoint.PLAYER)
```
should be done in `getStats()`/`getGames()`.